dummy-issuer/src/lib.rs

```diff
@@ -0,0 +1,236 @@
+use asset_util::CertifiedAssets;
+use candid::{candid_method, Principal};
+use canister_sig_util::signature_map::{SignatureMap, LABEL_SIG};
+use canister_sig_util::CanisterSigPublicKey;
+use ic_cdk::api::{set_certified_data, time};
+use ic_cdk_macros::{query, update};
+use ic_certification::{fork_hash, labeled_hash, Hash};
+use identity_core::convert::FromJson;
+use identity_credential::credential::Subject;
+use identity_credential::error::Error as JwtVcError;
+use identity_credential::validator::JwtValidationError;
+use identity_jose::jws::Decoder;
+use identity_jose::jwt::JwtClaims;
+use lazy_static::lazy_static;
+use serde_bytes::ByteBuf;
+use serde_json::Value;
+use sha2::{Digest, Sha256};
+use std::cell::RefCell;
+use vc_util::issuer_api::{
+    CredentialSpec, DerivationOriginData, DerivationOriginError, DerivationOriginRequest,
+    GetCredentialRequest, Icrc21ConsentInfo, Icrc21Error, Icrc21VcConsentMessageRequest,
+    IssueCredentialError, IssuedCredentialData, PrepareCredentialRequest, PreparedCredentialData,
+    SignedIdAlias,
+};
+use vc_util::{
+    build_credential_jwt, did_for_principal, vc_jwt_to_jws, vc_signing_input,
+    vc_signing_input_hash, CredentialParams,
+};
+
+const ISSUER_URL: &str = "https://dummy-issuer.vc";
+const CREDENTIAL_URL_PREFIX: &str = "data:text/plain;charset=UTF-8,";
+const MINUTE_NS: u64 = 60 * 1_000_000_000;
+// The expiration of issued verifiable credentials.
+const VC_EXPIRATION_PERIOD_NS: u64 = 15 * MINUTE_NS;
+
+thread_local! {
+    /// Non-stable structures
+    // Canister signatures
+    static SIGNATURES : RefCell<SignatureMap> = RefCell::new(SignatureMap::default());
+    // Assets for the management app
+    static ASSETS: RefCell<CertifiedAssets> = RefCell::new(CertifiedAssets::default());
```

Comment:
The dummy issuer does not serve http requests currently. Can this be removed then?

Reply:
Ah, yes, this was a question I had for you as well.

I copied some parts from the VC Playground and I needed this. I don't know whether I could accomplish the same without it.

The `ASSETS` are used to get the root hash. Can I get it somehow else?

```
fn update_root_hash() {
    SIGNATURES.with_borrow(|sigs| {
        ASSETS.with_borrow(|assets| {
            let prefixed_root_hash = fork_hash(
                // NB: Labels added in lexicographic order.
                &assets.root_hash(),
                &labeled_hash(LABEL_SIG, &sigs.root_hash()),
            );

            set_certified_data(&prefixed_root_hash[..]);
        })
    })
}
```

And:

```
    let sig_result = SIGNATURES.with(|sigs| {
        let sig_map = sigs.borrow();
        let certified_assets_root_hash = ASSETS.with_borrow(|assets| assets.root_hash());
        sig_map.get_signature_as_cbor(
            &CANISTER_SIG_SEED,
            message_hash,
            Some(certified_assets_root_hash),
        )
    });
```

Reply:
Yes, you can leave off the top-level fork and in the second snipped just supply `None`.

Reply:
@frederikrothenberger I tested it and it works. I also removed the asset_util dependency.